lib/saml/health_status.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+require 'saml/settings_service'
+require 'sentry_logging'
+
+module SAML
+  module StatusMessages
+    NOT_ATTEMPTED = 'SAML Metadata retrieve not yet attempted'
+    MISSING       = 'SAML Metadata retrieve attempted, but none returned'
+    CERT_INVALID  = 'IDP certificate is invalid'
+  end
+
+  class HealthStatus
+    class << self
+      include SentryLogging
+      include StatusMessages
+
+      attr_accessor :consecutive_signature_failures
+
+      def healthy?
+        error_message.empty?
+      end
+
+      def error_message
+        return StatusMessages::NOT_ATTEMPTED unless fetch_attempted?
+        return StatusMessages::MISSING       unless metadata_received?
+        return StatusMessages::CERT_INVALID  unless idp_cert_valid?
+        ''
+      end
+
+      private
+
+      def fetch_attempted?
+        SettingsService.fetch_attempted == true
```

Comment:
`fetch_attempted` may be `nil`.  I want this method to return only `true` or `false`